R/run-nmtran.R

```diff
@@ -0,0 +1,330 @@
+#' Interface for running `NM-TRAN` on model objects
+#'
+#' Function to run `NM-TRAN` on a model object to validate its control stream
+#'  for correct coding before submission. The `NM-TRAN` dataset (`FDATA`) and
+#'  other `NONMEM` artifacts can be further inspected by keeping the run directory
+#'  around.
+#'
+#' @details
+#' `NM-TRAN` is a preprocessor for `NONMEM` that translates user-specified
+#' control stream data and instructions into a form executable by `NONMEM`.
+#'
+#' Note that `nmtran_presort` is run ahead of `NM-TRAN` for `NONMEM` versions
+#' `'nm74gf'`, `'nm74gf_nmfe'`, and `'nm75'`.
+#'  - `nmtran_presort` is an supplementary utility that preprocesses the control
+#'     stream to ensure it is in the correct format for `NM-TRAN`. It is particularly
+#'     relevant for handling specific data manipulations and ensuring compatibility
+#'     with the `NM-TRAN` executable.
```

Comment:
> \`nmtran\_presort\` is an supplementary utility that preprocesses the control stream

`s/a/an/`

It's run ahead of NM-TRAN on the control stream, so all of that seems fair enough (though I don't think "supplementary" adds anything). All right.

> to ensure it is in the correct format for \`NM-TRAN\`. It is particularly relevant for handling specific data manipulations and ensuring compatibility with the \`NM-TRAN\` executable.

Interesting. What's the source of that information? I don't see presort mentioned in the NONMEM help pages or the NONMEM 7.5's NM-TRAN guide (IV). I took a glance at `/opt/NONMEM/nm75/util/nmtran_presort.f90`, but my Fortran skills are non-existent.

Reply:
I put that together from going through chapter 6 of the nonmem 7.4 guides and some of the appendices regarding `NM-TRAN` and preprocessing. I think it's conceptually related to the data preprocessing step referred to in that guide, though isnt included within that step itself (the guide never refers to this step directly). I tried to make that description somewhat vague, but perhaps it makes sense to remove without a specific source.

Reply:
> but perhaps it makes sense to remove without a specific source.

I would prefer to, yes.  It's enough to document that we run it to follow `nmfe*`.